bin/build_config_upload.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python
+#
+# This script is run by a cron job as the hwcron user
+#
+# Regularly checks a queue to rebuild assignment configurations for recently modified gradeables.
+#
+
+import os
+import pwd
+import time
+import subprocess
+import glob
+import json
+
+
+# ------------------------------------------------------------------------
+# this script is intended to be run only from the cron job of user hwcron
+username = pwd.getpwuid( os.getuid() ).pw_name
+if (username != "hwcron") :
+	print "ERROR!  This script must be run by hwcron"
+	exit()
+
+
+# ------------------------------------------------------------------------
+def buildOne(data) :
+	semester = data["semester"]
+	course = data["course"]
+	gradeable = data["gradeable"]
+
+	build_script = "/var/local/submitty/courses/" + semester + "/" + course + "/BUILD_" + course + ".sh"
+	build_output = "/var/local/submitty/courses/" + semester + "/" + course + "/build_script_output.txt"
+
+	f = open(build_output,'w')
+	subprocess.call([ build_script, gradeable ],stdout=f,stderr=f)
+
+
+# ------------------------------------------------------------------------
+def buildAll() :
+    for file in glob.glob("/var/local/submitty/to_be_built/*.json") :
```

Comment:
Please don't use `file` as a variable name as that's a builtin for python. `filename` perhaps?